.github/workflows/abicheck.yaml

```diff
@@ -0,0 +1,35 @@
+name: ABI Check
+
+on:
+  push:
+    branches:
+      - v2.*
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - v2.*
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:latest
+      options: --privileged  -v /usr/src:/usr/src -v /lib/modules:/lib/modules
+
+    steps:
+      - name: Setup
+        run: |
+          dnf install -y \
+            abi-compliance-checker git ninja-build meson pkg-config gcc \
+            libmicrohttpd-devel userspace-rcu-devel libyaml-devel lz4-devel \
+            libbsd-devel libcurl-devel libpmem-devel
+
+      - uses: actions/checkout@v2
+
+      - name: Initialize
+        run: |
+          dnf install -y findutils ShellCheck
+
+      - name: Lint
+        run: |
+          ./scripts/dev/shellcheck.sh
```

Comment:
Using shellcheck ?

Reply:
This was just me saving off my work so that I would remember to come back to it.